### PR TITLE
feat: add feature to reorder columns

### PR DESCRIPTION
**Pull request by ssd71**
- Allows reordering columns in the column toggle dropdown

Tested with changing streams and reordering columns multiple times.

Some concerns:
- The `react-beautiful-dnd` library adds some load time but it is one of the more stable dnd frameworks, I am open to any suggestions for any other libraries.

Some libraries I know about:
- `@dnd-kit`: Somewhat smaller
- `react-dnd`: Slightly smaller than `@dnd-kit`
- `react-grid-layout`: Size between `react-beautiful-dnd` and `@dnd-kit`, but can be considered if the project is expected to have resizable and/or movable widgets since it can also manage that

Let me know what you think

### CODE REVIEW

**Review by nitisht — APPROVED** [review]
LGTM 🚀 . Thank you for your hard work @ssd71

**Comment by nitisht** [comment]
Fixes #59

File: src/hooks/useGetLogStreamSchema.ts [hunk 1]
```diff
@@ -36,5 +37,33 @@ export const useGetLogStreamSchema = () => {
...
+
+				const newFields: Field[] = [...prev.fields];
+				for (let i = 0; i < prev.fields.length; i++) {
+					let offset = 0;
+					if (source < destination && i >= source && i < destination) offset = 1;
+					else if (destination < source && i > destination && i <= source) offset = -1;
+					newFields[i] = prev.fields[i + offset];
+				}
+				newFields[destination] = prev.fields[source];
+				debugger;
```
Comment (Ryiski) [inline src/hooks/useGetLogStreamSchema.ts:57]: remove debugger

File: src/pages/Logs/LogTable.tsx [hunk 2]
```diff
@@ -285,23 +289,50 @@ const ThColumnMenu: FC<ThColumnMenuProps> = (props) => {
 							<IconDotsVertical size={px('1.2rem')} />
 						</ActionIcon>
 					</Menu.Target>
-				</Center>
-				<Menu.Dropdown className={thColumnMenuDropdown}>
-					{logSchemaFields.map((field) => {
+				</Center>    
+	<DragDropContext
+      onDragEnd={({ destination, source }) => {
+		// debugger;
```
Comment (Ryiski) [inline src/pages/Logs/LogTable.tsx:295]: clean up

File: src/hooks/useGetLogStreamSchema.ts (hunk 1 above)
Comment (ssd71) [reply src/hooks/useGetLogStreamSchema.ts]: Removed

File: src/pages/Logs/LogTable.tsx (hunk 2 above)
Comment (ssd71) [reply src/pages/Logs/LogTable.tsx]: I believe it's done, please check again

**Review by kartik-gupta-ij — APPROVED** [review]

**Pull request merged**